pkg/runtime/openshell/gateway.go

```
@@ -0,0 +1,331 @@
+	defer cancel()
+
+	_, err := r.executor.Output(checkCtx, io.Discard, "sandbox", "list", "--names")
+	return err == nil
+}
+
+// stopGateway terminates the running gateway process and cleans up state files.
+func (r *openshellRuntime) stopGateway(ctx context.Context) error {
+	state, err := loadGatewayState(r.storageDir)
+	if err != nil || state.PID <= 0 {
+		removeGatewayState(r.storageDir)
+		return nil
```

golangci-lint (2.12.1)

[error] error is not nil (line 56) but it returns nil

(nilerr)